lib/samplers/ratio-based-sampler.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2025 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+const Sampler = require('./sampler')
+
+/**
+ * Trace ID ratio-based sampler adapted for New Relic agent use,
+ * based on OpenTelemetry's sampler of the same name.
+ */
+class TraceIdRatioBasedSampler extends Sampler {
+  /**
+   * @param {object} opts Sampler options.
+   * @param {Agent} opts.agent - The New Relic agent instance.
+   * @param {number} opts.ratio - The sampling ratio (0 to 1).
+   */
+  constructor(opts) {
+    super() // no-op
+    if (!opts.agent) {
+      throw new Error('must have an agent instance')
+    }
+    this._ratio = this._normalize(opts.ratio)
+    this._upperBound = Math.floor(this._ratio * 0xffffffff)
+  }
+
+  applySamplingDecision({ transaction }) {
+    if (!transaction) return
+    // eslint-disable-next-line sonarjs/pseudo-random
+    const initPriority = Math.random()
+    transaction.sampled = this.shouldSample(transaction.traceId)
+    transaction.priority = transaction.sampled ? initPriority + 1 : initPriority
+    // Truncate the priority after potentially modifying it to avoid floating
+    // point errors.
+    transaction.priority = ((transaction.priority * 1e6) | 0) / 1e6
+  }
+
+  /**
+   *
+   * @param {string} traceId transaction trace id
+   * @returns {boolean} whether to sample a transaction based on given trace id
+   */
+  shouldSample(traceId) {
+    const accumulated = this._accumulate(traceId)
+    return accumulated <= this._upperBound
+  }
+
+  toString() {
+    return 'TraceIdRatioBasedSampler'
+  }
+
+  /**
+   * Normalizes the sampling ratio.
+   * @param {number} ratio The ratio provided by the config.
+   * @returns {number} The normalized ratio [0, 1].
+   */
+  _normalize(ratio) {
```

Comment:
i know this is copying the otel sampler but as i stated above, it should do this normalization when the config is constructed